app/src/org/commcare/tasks/SendTask.java

```diff
@@ -91,6 +94,26 @@ protected Boolean doTaskBackground(Void... params) {
                 continue;
             }
             try {
+
+                // see if we have a form.properties file to load the PostURL from
+                FilenameFilter filter = new FilenameFilter() {
+                    @Override
+                    public boolean accept(File dir, String filename) {
+                        return filename.equals(ZipTask.FORM_PROPERTIES_FILE);
+                    }
+                };
+
+                File[] fileContents = f.listFiles(filter);

```

Comment:
I would assume that `f` stands for 'file', but here you are calling a directory specific method `listFiles` on it. Would be helpful to rename `f` to `dir` or `formInstanceDir` (if that is an accurate description).